java/connector-node/risingwave-sink-deltalake/src/main/java/com/risingwave/connector/DeltaLakeSinkConfig.java

```diff
@@ -16,43 +16,20 @@
 
 package com.risingwave.connector;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.risingwave.connector.api.sink.CommonSinkConfig;
+import com.risingwave.connector.common.S3Config;
 
-public class DeltaLakeSinkConfig extends CommonSinkConfig {
+public class DeltaLakeSinkConfig extends S3Config {
+    @JsonProperty(value = "location")
     private String location;
 
-    private String locationType;
-
+    @JsonProperty(value = "type")
     private String sinkType;
 
     @JsonProperty(value = "force_append_only")
     private Boolean forceAppendOnly;
 
-    @JsonCreator
-    public DeltaLakeSinkConfig(
-            @JsonProperty(value = "location") String location,
-            @JsonProperty(value = "location.type") String locationType,
-            @JsonProperty(value = "type") String sinkType) {
-        this.location = location;
-        this.locationType = locationType;
-        this.sinkType = sinkType;
-    }
```

Comment:
I have recovered to use the constructor to enforce the non-null requirement. 

However, I think the current way to enforce non-null requirement is too implicit. 

Shall we investigate on some proper way to enforce this non-null requirement? Such as using NotNull annotation, or we introduce a inheritable method named like `checkField` and each sub class of `CommonConfig` can define their own logic of config checking.

Besides, I notice that the current `primary_key` and `type` (the sink type) are defined in each sub class of `CommonConfig`. However, the two field are actually used our sink framework, which means these two fields are built in config, just like the `connector`. Therefore, we shall consider moving them to `CommonConfig`.

Reply:
>  Shall we investigate on some proper way to enforce this non-null requirement?

+1. We should find a better way to enforce the `required` and `optional` constraints.

> Besides, I notice that the current primary_key and type (the sink type) are defined in each sub class of CommonConfig. However, the two field are actually used our sink framework, which means these two fields are built in config, just like the connector. Therefore, we shall consider moving them to CommonConfig.

Currently all fields in `CommonConfig` are mandatory (`required`). I feel the two fields should be included in `CommonConfig` logically as well. This can be done together with the above improvement.